hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/DeleteHoodieTableCommand.scala

```diff
@@ -74,4 +88,13 @@ case class DeleteHoodieTableCommand(dft: DeleteFromTable) extends HoodieLeafRunn
 
     Seq.empty[Row]
   }
+
+  def tryPruningDeleteRecordSchema(query: LogicalPlan, requiredColNames: Seq[String]): LogicalPlan = {
+    val filteredOutput = query.output.filter(attr => isMetaField(attr.name) || requiredColNames.contains(attr.name))
+    if (filteredOutput == query.output) {
+      query
+    } else {
+      Project(filteredOutput, query)
+    }
```

Comment:
Can you elaborate what kind of `InternalRow` this project could produce?

Reply:
> Can you elaborate what kind of `InternalRow` this project could produce?

`GenericRowWithSchema`

Reply:
Does the schema change after the projection?

Reply:
> Does the schema change after the projection?

This ends up generating only the record with schema in `project`

Reply:
Does the delete records handler follow this contract then? Do they use the right schema?

Reply:
> Does the delete records handler follow this contract then? Do they use the right schema?

In subsequent processing, there is no relationship with schema.
<img width="1464" alt="image" src="https://github.com/user-attachments/assets/ec35c03c-3a9e-44db-bc49-d1fca8fe1d15" />